server/BraveAlerterConfigurator.js

```diff
@@ -0,0 +1,126 @@
+const { BraveAlerter, AlertSession, ALERT_STATE } = require('brave-alert-lib')
+const db = require('./db/db.js')
+
+class BraveAlerterConfigurator {
+    createBraveAlerter() {
+        return new BraveAlerter(
+            this.getAlertSession.bind(this),
+            this.getAlertSessionByPhoneNumber.bind(this),
+            this.alertSessionChangedCallback,
+            true,
+            this.getReturnMessage.bind(this),
+        )
+    }
+
+    async getAlertSession(sessionId) {
+        const session = await db.getSessionWithSessionId(sessionId)
+        const installation = await db.getInstallationWithInstallationId(session.installationId)
+    
+        const incidentCategoryKeys = this.createIncidentCategoryKeys(installation.incidentCategories)
+    
+        let alertSession = new AlertSession(
+            session.id,
+            session.state,
+            session.incidentType,
+            session.notes,
+            `There has been a request for help from Unit ${session.unit} . Please respond "Ok" when you have followed up on the call.`,
+            installation.responderPhoneNumber,
+            incidentCategoryKeys,
+            installation.incidentCategories,
+        )
+    
+        return alertSession
+    }
+
+    async getAlertSessionByPhoneNumber(toPhoneNumber) {
+        const session = await db.getMostRecentIncompleteSessionWithPhoneNumber(toPhoneNumber)
+        const installation = await db.getInstallationWithInstallationId(session.installationId)
+    
+        const incidentCategoryKeys = this.createIncidentCategoryKeys(installation.incidentCategories)
+    
+        let alertSession = new AlertSession(
+            session.id,
+            session.state,
+            session.incidentType,
+            session.notes,
+            `There has been a request for help from Unit ${session.unit} . Please respond "Ok" when you have followed up on the call.`,
+            installation.responderPhoneNumber,
+            incidentCategoryKeys,
+            installation.incidentCategories,
+        )
+    
+        return alertSession
+    }
+
+    async alertSessionChangedCallback(alertSession) {
```

Comment:
I'm confused about this and the associated addition of the db methods.

Is there a reason we can't use a single call to `db.saveSession` here? This would cut down on the number of db operations and require less new code.

Maybe I'm missing something though.

Reply:
Good question.

It's because of how the library is using it. Right now it's only setting the values of `alertSession` that it wants to change. For example, when it gets back any details/notes, then `alertSession.alertState` would be set to `COMPLETED` and `alertSession.details` would be set to, say, `'All good now'`. But it would leave the `alertSession.incidentCategory` as `undefined`. So if I were to use `db.saveSession` then I would overwrite `incidentCategory` in the DB with `null`, which is not what we want.

I agree that this is confusing. A couple of ways to change it so that it might be clearer would be:
* to have this function take in separate parameters for the values that can change instead of an incomplete `alertSession` object and otherwise leave the functionality the same with the separate DB calls.
* to have the library send back the whole session and then use `db.saveSession`; however, this means the library would need to learn how to do transactions using these callback functions (so there isn't a race condition between retrieving the values in the DB vs setting them to the new values).

What do you think?

Reply:
Change this to a transaction that 
1. Reads the session
2. Writes the whole session with the changes from `alertSession`

Reply:
Done.

Reply:
@trisapeace thanks.